platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/signed/SignedState.java

```diff
@@ -755,16 +718,15 @@ public void pruneInvalidSignatures(@NonNull final Roster trustedRoster) {
             sigSet.removeSignature(nodeId);
         }
 
-        long newWeight = 0;
+        // Recalculate signing weight. We should do this even if we don't remove signatures.
+        signingWeight = 0;
 
         for (final NodeId nodeId : sigSet) {
             final RosterEntry entry = entriesByNodeId.get(nodeId.id());
             if (entry != null) {
-                newWeight += entry.weight();
+                signingWeight += entry.weight();
             }
         }
-
-        signingWeight = newWeight;
     }
```

Comment:
Minor: please note that the old code would only update the `signingWeight` instance variable if this code never threw any exceptions. Not sure if there's any code that relies on this fact, but your version changes this invariant. I'm totally fine with keeping your version as it avoids an extra long value on the stack, but ideally it would be great to confirm if this is safe to do.

Reply:
Are you referring to the old code in this method? If so, this is all new code that I added in a previous PR. I added the local `long` variable and then did the update to `signingWeight` in one operation. This was _not_ the same as the previous `pruneInvalidSignatures(AddressBook)`, but in this PR I did "revert" my change to mirror what was done in the original prune method that used the address book.

I can absolutely revert this back to my original code (single update), but it wouldn't match the original prune method. I made this change because I didn't know if anything was dependent on the original way of doing the recalculation. :)

Reply:
Ah, I see. Thanks for clarifying. This indicates that this is safe then. I'm good with the current version of the code.